homework/Repository.js

```diff
@@ -8,8 +8,8 @@ class Repository {
     this.repository = repository;
   }
 
-  detectRepositoryAndValue(repository, value) {
-    if (typeof repository !== 'undefined' && value !== null) {
+  detectValueOfRepository(value) {
```

Comment:
If you don't mind my saying so, this is a completely useless method. If the only thing you want to do is check whether a value is null or not, surely you don't need a function for that.

Reply:
I have changed already that method in last changes , After last changes.. that method is checking that İf value of repository is equal to null or not

Reply:
I still do not think this function adds value. Firstly, its name pretends to do more than it actually does. If you look at the function body, you can find that it only tests whether the value passed to it as a parameter is `null` or not and returns a boolean value to indicate the result. Based on what is actually does this function is better called `isNotNull(value)`. It is not at all exclusive to working with repositories as its name suggests.

Secondly, this function does trivial work. You could rewrite as:

```js
function isNotNull(value) {
  return value != null;
}
```

This is not worth to write a function for. Just use the test for `null` directly, e.g.:

```js
if (selectedRepository.name != null) {
```